docs/concepts/how-ipfs-works.md

```diff
@@ -5,86 +5,168 @@ description: Learn how the InterPlanetary File System (IPFS) works and why it's
+:::callout
+**Learn more**
 
-## Distributed hash tables (DHTs)
+Want learn more about Bitswap? See the [Bitswap deep dive](../concepts/bitswap.md).
+:::
+
+### mDNS
+
+To quickly and efficiently discover peers, IPFS uses Multicast Domain Name System (mDNS), a type of <VueCustomTooltip label="A system in which human-readable internet domain names are mapped to IP addresses. The Domain Name System has been an essential component of the internet since 1985." underlined multiline is-medium>DNS</VueCustomTooltip> protocol that resolves human-readable internet domain names to IP names without the use of a <VueCustomTooltip label="Any computer application that implements a system in which human-readable internet domain names are mapped to IP addresses (DNS)." underlined multiline is-medium>name server</VueCustomTooltip>. The process for peer discovery is as follows:
+
+1. Given a peer ID, a node broadcasts a query message to multiple peers in the network, asking that the peer having the given peer ID to identify itself.
+2. The peer with the given peer ID then broadcasts a message with its IP address to other peers on the network, which can then use that information to update their own mDNS records.
+
+The use of mDNS in IPFS has several benefits that make it a clear choice for peer discovery over traditional DNS:
+
+- Nodes can broadcast messages requesting and reporting address information to multiple peers, increasing speed and efficiency of peer discovery.
+- Nodes do not rely on a centralized name server to discover other peers, improving network resilience, decentralization and censorship resistance.
 
-To find which peers are hosting the content you're after (_discovery_), IPFS uses a [distributed hash table](dht.md), or DHT. A hash table is a database of keys to values. A _distributed_ hash table is one where the table is split across all the peers in a distributed network. To find content, you ask these peers.
+### Delegated routing over HTTP
 
-The [libp2p project](https://libp2p.io/) is the part of the IPFS ecosystem that provides the DHT and handles peers connecting and talking to each other. (Note that, as with IPLD, libp2p can also be used as a tool for other distributed systems, not just IPFS.)
+Delegated content routing is a mechanism for IPFS implementations to use for offloading content routing to another process/server using an HTTP API. For example, if an IPFS node does not implement the DHT, a delegated router can search the DHT for peers on its behalf. The main benefit of delegated routing is that nodes are not required to implement routing functionality themselves if they do not have the computing resources to do so, or wish to build an IPFS system with a custom backend for routing. Thus, delegated routing over HTTPS provides IPFS nodes with a standard interface that allows more flexibility in terms of how content routing works. 
```

Comment:
Link to the spec here: https://github.com/ipfs/specs/blob/main/routing/DELEGATED_CONTENT_ROUTING_HTTP.md